synapse/rest/client/receipts.py

```diff
@@ -56,8 +56,8 @@ async def on_POST(
     ) -> Tuple[int, JsonDict]:
         requester = await self.auth.get_user_by_req(request)
 
-        if not room_id or not event_id:
-            raise SynapseError(400, "A room ID and event ID must be specified")
+        if not RoomID.is_valid(room_id) or not event_id.startswith(EventID.SIGIL):
```

Comment:
Not super happy with the `startswith` but `EventID.is_valid` is broken for non-V1/2 event IDs so unsure how to proceed.

Reply:
Filed #14638.